### PR TITLE
Task 4 - GET api/articles/:article_id

### PR DESCRIPTION
**Pull request by KDBlockdev**
implementing get article by article id, test currently passing

### CODE REVIEW

File: __tests__/news.test.js [hunk 1]
```diff
@@ -48,3 +48,51 @@ describe("/api/topics", () => {
     });
   });
 });
+
+//Test 3 - Get all articles
+describe("/api/articles/", () => {
+  describe("GET Article object", () => {
```
Comment (hevdevs) [inline __tests__/news.test.js]: get all articles is ticket 8, i know it seems harsh but we want to be tackling one ticket at a time so let's get rid of this for now

File: __tests__/news.test.js [hunk 2]
```diff
@@ -48,3 +48,51 @@ describe("/api/topics", () => {
...
+  describe("GET Article object by specific article ID", () => {
+    test("200: responds with an Object of a specific article", () => {
+      return request(app)
+        .get("/api/articles?article_id=1")
+        .expect(200)
+        .then(({ body }) => {
+          const articles = body;
+          expect(articles).toBeInstanceOf(Object);
+          expect.objectContaining({
+            author: expect.any(String),
```
Comment (hevdevs) [inline __tests__/news.test.js]: if we are requesting an article with a specific id we don't want to be expecting any value, we are expecting the specific information relevant to that article so need to adjust these tests to match that

File: controllers/news.controllers.js [hunk 3]
```diff
@@ -9,7 +10,21 @@ const getTopics = (req, res) => {
   collectTopics().then((topics) => res.status(200).send(topics));
 };
 
+const getArticles = (req, res) => {
+  collectArticles().then((articles) => res.status(200).send(articles));
+  // console.log(articles)
```
Comment (hevdevs) [inline controllers/news.controllers.js]: remove cl before merge

**Review by hevdevs — CHANGES_REQUESTED** [review]
some changes to make here Kieron!
we also need some basic error handling tests here: what happens if the article number does not exist i.e. we try to access /reviews/9999? what happens if we try to access a review id with something other than a number i.e. an endpoint of /reviews/pineapple?

File: models/news.models.js [hunk 4]
```diff
@@ -1,5 +1,10 @@
-const db = require('../db/connection.js');
+const db = require("../db/connection");
 
 exports.collectTopics = () => {
   return db.query("SELECT * FROM topics;").then((result) => result.rows);
 };
+
+exports.collectArticleById = (article_id) => {
+    return db.query("SELECT * FROM articles WHERE article_id = $1;", [article_id]).then((result) => result.rows[0]);
```
Comment (pkaodev) [inline models/news.models.js]: splitting this across multiple lines will make it more readable

File: __tests__/news.test.js [hunk 5]
```diff
@@ -36,6 +36,7 @@ describe("/api/topics", () => {
         .then(({ body }) => {
           const topics = body;
```
Comment (pkaodev) [inline __tests__/news.test.js:37]: your data should be sent back as a key on body.  double check that you're running your tests before commiting.

File: __tests__/news.test.js [hunk 6]
```diff
@@ -48,3 +49,35 @@ describe("/api/topics", () => {
...
+
+//Test 3 - Get articles by ID
+describe("/api/articles/:article_id", () => {
+  describe("GET Article object by specific article ID", () => {
+    test("200: responds with an Object of a specific article", () => {
+      return request(app)
+        .get("/api/articles/1")
+        .expect(200)
+        .then(({ body }) => {
+          const articles = body;
```
Comment (pkaodev) [inline __tests__/news.test.js]: from your controller it looks like you are sending back the data on body.article.  double check that you are running your tests before commiting.

File: __tests__/news.test.js [hunk 7]
```diff
@@ -48,3 +49,35 @@ describe("/api/topics", () => {
...
+          expect(articles).toBeInstanceOf(Object);
```
Comment (pkaodev) [inline __tests__/news.test.js]: this is not needed as you are inferring that you are receiving an object back on the next line.

File: __tests__/news.test.js [hunk 8]
```diff
@@ -48,3 +49,35 @@ describe("/api/topics", () => {
...
+            body: "I find this existence challenging",
+            topic: "mitch",
+            created_at: 1594329060000,
+            votes: 100,
+          });
+        });
+    });
+    test("status:404, responds with an error message when passed a bad article ID e.g. not a number", () => {
+      return request(app)
+        .get("/api/articles?article_id=dog")
```
Comment (pkaodev) [inline __tests__/news.test.js]: this test is incorrect, this is not the right endpoint (this is /api/articles)

File: __tests__/news.test.js [hunk 9]
```diff
@@ -48,3 +49,35 @@ describe("/api/topics", () => {
...
+        .expect(404)
```
Comment (pkaodev) [inline __tests__/news.test.js:83]: This would be a 400, not a 404

File: __tests__/news.test.js [hunk 10]
```diff
@@ -48,3 +49,35 @@ describe("/api/topics", () => {
...
+        .then(({ body }) => {
+          expect(body.msg).toBe(undefined, "<<< route does not exist");
```
Comment (pkaodev) [inline __tests__/news.test.js]: make sure you are sending back a helpful message, not undefined.

**Comment by pkaodev** [comment]
Check the comments I wrote.

There are changes to be made in the test, and some more tests to be written (including errors).

If you have any questions please message me and we can have a chat.

Send another nchelp PR when you have finished.

File: __tests__/news.test.js [hunk 11]
```diff
@@ -48,3 +49,41 @@ describe("/api/topics", () => {
...
+        .then(() => {
+          expect.objectContaining({
+            author: "butter_bridge",
+            title: "Living in the shadow of a great man",
+            article_id: 1,
+            body: "I find this existence challenging",
+            topic: "mitch",
+            created_at: 1594329060000,
+            votes: 100,
+          });
```
Comment (douglashellowell) [inline __tests__/news.test.js]: *what* are you expecting to be this object? 🤔

You're getting a false-positive here

You **must** see a failing test before you solve it!

**Review by pkaodev — APPROVED** [review]
Looks good, merge away

**Pull request merged**